### ISSUE TITLE
mfem::Array<T> rule of 5 methods don't check if typename T is plain old data 

### ISSUE THREAD

**Issue by goxberry** [mfem/mfem#1648]
The `mfem::Array<T>` documentation states that `typename T` must be plain old data. As of C++11, this condition is testable via the `<type_traits>` header, e.g.:

```c++
#include <type_traits>
#include <iostream>

// std::is_pod<T> also works, but is deprecated as of C++20
if (std::is_trivial<T> && std::is_standard_layout<T>)
{
    std::cout << "Type is plain old data" << std::endl;
}
else
{
    std::cout << "Type is not plain old data" << std::endl;
}
```

A code snippet like this could be used to enforce the design contract declared in the documentation, assuming the project is amenable to the idea.

**Comment by tzanio** [comment]
I am personally don't have strong feelings one way or the other about this -- @v-dobrev and @jakubcerveny, what do you think?

**Comment by v-dobrev** [comment]
We can these checks (and the move constructor+assignment). One option is to add `static_asset`s; another is to use some `enable_if` "magic". What did you have in mind, @goxberry?

**Comment by goxberry** [comment]
I’d favor `enable_if` because it’s a compile-time check. I was originally thinking something like `static_assert`, but I like @v-dobrev’s suggestion better, which taught me something new. Thanks @v-dobrev !

**Comment by jakubcerveny** [comment]
I think `static_assert` would work, it is compile-time and also simple/clear.

**Comment by tzanio** [comment]
I have a slight preference for `static_assert`.

**Comment by tzanio** [comment]
@goxberry, do you want to propose a PR for this?

**Comment by tzanio** [comment]
ping: @goxberry

**Comment by v-dobrev** [comment]
This is being addressed in #2214. Let's move any further discussion there.
